Tweeter.Tests/TweeterRepoTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Tweeter.DAL;
+using Moq;
+using Tweeter.Models;
+using System.Data.Entity;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Tweeter.Tests
+{
+    [TestClass]
+    public class TweeterRepoTests
+    {
+
+        private Mock<TweeterContext> MockContext;
+        private Mock<DbSet<Twit>> TweeterUsers;
+        private List<Twit> twits;
+
+        [TestInitialize]
+        public void Setup()
+        {
+            MockContext = new Mock<TweeterContext>();
+            TweeterUsers = new Mock<DbSet<Twit>>();
+
+
+            MockContext.Setup(x => x.TweeterUsers).Returns(TweeterUsers.Object);
+
+            twits = new List<Twit>
+            {
+                new Twit
+                {
+                   Username = "user-1"
+                },
+                new Twit
+                {
+                   Username = "user-2"
+                }
+            };
+
+            ConnectMocksToDatastore();
+        }
+
+        public void TestCleanup()
+        {
+            MockContext = null;
+            TweeterUsers = null;
+            twits = null;
+        }
+
+
+        public void ConnectMocksToDatastore()
+        {
+            var queryableTwits = twits.AsQueryable();
+            TweeterUsers.As<IQueryable<Twit>>().Setup(x => x.Provider).Returns(queryableTwits.Provider);
+            TweeterUsers.As<IQueryable<Twit>>().Setup(x => x.Expression).Returns(queryableTwits.Expression);
+            TweeterUsers.As<IQueryable<Twit>>().Setup(x => x.ElementType).Returns(queryableTwits.ElementType);
+            TweeterUsers.As<IQueryable<Twit>>().Setup(x => x.GetEnumerator()).Returns(queryableTwits.GetEnumerator());
```

Comment:
This is missing mappings to return Tweets when you iterate over it, but I've seen them added in your later code